openwebstart/src/main/java/com/openwebstart/proxy/WebStartProxySelector.java

```diff
@@ -15,13 +15,16 @@
 import java.net.SocketAddress;
 import java.net.URI;
 import java.util.List;
+import java.util.stream.Collectors;
 
 public class WebStartProxySelector extends ProxySelector {
 
     private static final Logger LOG = LoggerFactory.getLogger(WebStartProxySelector.class);
 
     private final ProxyProvider proxyProvider;
 
+    private List<Proxy> proxyList;
```

Comment:
This class is used by multiple threads. Therefore it must not save context which is only valid for a single thread.

Reply:
Right. I had some doubt about this. Http requests are made in parallel. There can be different Urls in the jnlp file and it is quite possible for different proxy list is specified in PAC script for different urls Alternative is to use call select every time.